In [1]:
import pandas as pd
from pytesseract import pytesseract
import cv2
import numpy as np
from typing import List, Tuple
import matplotlib.pyplot as plt
from PIL import Image

import os
import sys

pytesseract.tesseract_cmd = r'/usr/bin/tesseract'

In [2]:
sys.path.append("../scripts")
from load_data import Loader

loader = Loader()

In [3]:
def get_files_name(directory:str, filter_extension:list=None)->list:
    
    # directory = f'/home/amanuel_zewdu/creative_image_optimization/data/Challenge_Data/Assets/{directory}/'
    
    files = []
    for filename in os.listdir(directory):
        f = os.path.join(directory, filename)
        
        # checking if it is a file
        if filter_extension != None:
            if os.path.isfile(f):
                if filename.split('.')[-1].upper() in filter_extension:
                    files.append(filename)
        else:
            if os.path.isfile(f):
                files.append(filename)

    return files

In [4]:
bucket = "s3://10ac-batch-6/data/w11/Challenge_Data.zip"
file_path = "Challenge_Data/performance_data.csv"

df = loader.load_csv(bucket,file_path)

In [5]:
logos = df.game_id.apply(lambda x:get_files_name(f'/home/amanuel_zewdu/creative_image_optimization/data/Challenge_Data/Assets/{x}/'))

In [6]:
logos_df = pd.DataFrame(logos)

In [8]:
logos_df['concat'] = logos_df.game_id.apply(lambda x: " ".join(x))

In [9]:
logos_df.head()

game_id  \
1  [endtext.png, landing_2.jpg, video.mp4, end-ha...   
2  [landing_2.jpg, video.mp4, end-mtlaurel.png, e...   
3  [endtext.png, thumbnail.jpg, landing_2.jpg, en...   
4  [logo.png, game_2.png, cta.png, engagement_ani...   
5  [logo.png, cta.png, shade.png, engagement_anim...   

                                              concat  
1  endtext.png landing_2.jpg video.mp4 end-haverf...  
2  landing_2.jpg video.mp4 end-mtlaurel.png end-h...  
3  endtext.png thumbnail.jpg landing_2.jpg endfra...  
4  logo.png game_2.png cta.png engagement_animati...  
5  logo.png cta.png shade.png engagement_animatio...

In [25]:
logos_df.concat.str.contains("logo").value_counts()

False    474
True     431
Name: concat, dtype: int64

only 431 creative assets directory contains information about logo